H/HDF5/build_tarballs.jl

```diff
@@ -136,6 +99,8 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("Zlib_jll"),
+    Dependency("OpenSSL_jll"),
```

Comment:
OpenSSL and LibCurl are apparently only needed for the aarch64 build (maybe also the apple build, but inspecting the shared lib it looked ok without them)